From patchwork Fri Jun  3 09:26:05 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Robinson <pbrobinson@gmail.com>
X-Patchwork-Id: 12868856
Return-Path: 
 <linux-arm-kernel-bounces+linux-arm-kernel=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id A9620C433EF
	for <linux-arm-kernel@archiver.kernel.org>;
 Fri,  3 Jun 2022 09:27:56 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=QQB2PsWIeCogVjFeFStUJ0naLVfs+V8tOubglW46un4=; b=omHQ6YmnN3jbFK
	UhUIRLbHxrSteaK+hT1svftgn+1OGMyAORTEF6MpNmQqKAIiIGJPCtvJ1N4muFw2VW310g8M/IPYv
	lW7EhAt3puV/NrwE2ftBe92gvnjCGwL1hLMKbyRah5XHWIStmuu7HymhCHZo7xa34HbrBJp12nZ17
	8ATdkd3exQ8v7R4ckSlhQ+0up3DQ9EOHVhU6uP/Ie6j12uX7+LH+LbYhKUj1EF6f7m7MmBSNBQhgv
	b3mQjoGr+XFFouLQ9YknqAIxHFYJtXwSGRvKCl4n8NkOS9ysVdr6hJlMQsjRuDJ+/yxR55ALa+YHg
	d5ZXfuRti0EsJ1qqaY3Q==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nx3Zw-006oD4-GN; Fri, 03 Jun 2022 09:26:48 +0000
Received: from mail-ej1-x62d.google.com ([2a00:1450:4864:20::62d])
	by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nx3ZU-006nts-92; Fri, 03 Jun 2022 09:26:21 +0000
Received: by mail-ej1-x62d.google.com with SMTP id me5so14270269ejb.2;
        Fri, 03 Jun 2022 02:26:16 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20210112;
        h=from:to:cc:subject:date:message-id:in-reply-to:references
         :mime-version:content-transfer-encoding;
        bh=r779VgB4wPd+dO4BnumFEEMoK5PeDls8DqXi+jzyZUk=;
        b=f+R+5qwCa/e3GQGG5kIjgqbEG8D0AdLZlPvlFWMh0nANKNQzW25nh67lFvGWHCmAme
         wOZoM1OGTH0JeDjUwMxkml0tO4CqWLYLZQ1j1e1/7UxPc+laMO9d2p8ZlhqXiAl8w5zz
         mOydMzVojO09VGYCR76P1s6s98gZneDmcc284F7lteaHR0iVxd9C48JjrqtvhutP+tEf
         VhR8yS0bB/ViUYgebS/icgc3Eqq79B+9kKvJAvRRdtBV6hyGOgfJn7mJEtBCcKKvJLkK
         Q8iICT4ObJzmFPSSfmdc75WMslEfDLvDI0zORjM9sprETND4l+SiBcZHEnH1vLKH/XT4
         bTdQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20210112;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=r779VgB4wPd+dO4BnumFEEMoK5PeDls8DqXi+jzyZUk=;
        b=Uo40ZEl8+d6ULz6fumaX87wkXkVIMrvD42EgcJQ0JzowPjmKDdEa9Izd5GllgSRWxR
         RFVYhfyIU2RV3We16UuVh1djU4Dpa9yRLDF4zJR3NwXj0euPaQkQv53WDJJSUPVjilD9
         xOnOCImvgt/nJB3IyZMLL1WZS+cxHibrxd9vK16yEjCP87pTpQIOoOOFB4ODwNsIMtjM
         PtTWyxikrHYei0HbU4qtPkGjc9N9Oyoyyer+ETAC4yUDOAxWv5fndcPQokdAmUJvlJn+
         RYHQtensFm1Zx2ankM2Qtd7UY3Lb6O5XF21z788EQx+ULVlPAPPwdDYSBnBfeHESWulY
         FArQ==
X-Gm-Message-State: AOAM533mfWLa6O7TyloMvMAR26Rh9sQtMC/3/uoXCliQvRY4JSiMWrdq
	mtcqNzHBD0mX1QVS7scj164=
X-Google-Smtp-Source: 
 ABdhPJymvabgWIehxXXv6/WYRjeKoxF/7tX8NhrdZc6D5mjdvNWIrrXXtNRW0Oi0fIl8Utj3MhmLSA==
X-Received: by 2002:a17:907:3e15:b0:6fe:f10e:6337 with SMTP id
 hp21-20020a1709073e1500b006fef10e6337mr7918695ejc.209.1654248375876;
        Fri, 03 Jun 2022 02:26:15 -0700 (PDT)
Received: from morpheus.home.roving-it.com
 (3.e.2.0.0.0.0.0.0.0.0.0.0.0.0.0.1.8.6.2.1.1.b.f.0.b.8.0.1.0.0.2.ip6.arpa.
 [2001:8b0:fb11:2681::2e3])
        by smtp.googlemail.com with ESMTPSA id
 b5-20020a17090691c500b006feaa22e367sm2617672ejx.165.2022.06.03.02.26.14
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Fri, 03 Jun 2022 02:26:15 -0700 (PDT)
From: Peter Robinson <pbrobinson@gmail.com>
To: Arnd Bergmann <arnd@arndb.de>,
	bcm-kernel-feedback-list@broadcom.com,
	Daniel Vetter <daniel@ffwll.ch>,
	David Airlie <airlied@linux.ie>,
	devicetree@vger.kernel.org,
	dri-devel@lists.freedesktop.org,
	Emma Anholt <emma@anholt.net>,
	Florian Fainelli <f.fainelli@gmail.com>,
	javierm@redhat.com,
	linux-arm-kernel@lists.infradead.org,
	linux-rpi-kernel@lists.infradead.org,
	maxime@cerno.tech,
	Nicolas Saenz Julienne <nsaenz@kernel.org>,
	Rob Herring <robh+dt@kernel.org>,
	Stefan Wahren <stefan.wahren@i2se.com>,
	Catalin Marinas <catalin.marinas@arm.com>,
	Will Deacon <will@kernel.org>
Cc: Peter Robinson <pbrobinson@gmail.com>,
	Nicolas Saenz Julienne <nsaenzjulienne@suse.de>,
	Rob Herring <robh@kernel.org>
Subject: [PATCH v6 1/6] dt-bindings: gpu: v3d: Add BCM2711's compatible
Date: Fri,  3 Jun 2022 10:26:05 +0100
Message-Id: <20220603092610.1909675-2-pbrobinson@gmail.com>
X-Mailer: git-send-email 2.36.1
In-Reply-To: <20220603092610.1909675-1-pbrobinson@gmail.com>
References: <20220603092610.1909675-1-pbrobinson@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20220603_022620_382025_873BE6B2 
X-CRM114-Status: GOOD (  10.20  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+linux-arm-kernel=archiver.kernel.org@lists.infradead.org

BCM2711, Raspberry Pi 4's SoC, contains a V3D core. So add its specific
compatible to the bindings.

Signed-off-by: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>
Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
Reviewed-by: Stefan Wahren <stefan.wahren@i2se.com>
Reviewed-by: Javier Martinez Canillas <javierm@redhat.com>
Acked-by: Rob Herring <robh@kernel.org>
---
Changes since v4:
- Change compatible to align downstream and other HW, reorder to suit

 Documentation/devicetree/bindings/gpu/brcm,bcm-v3d.yaml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Documentation/devicetree/bindings/gpu/brcm,bcm-v3d.yaml b/Documentation/devicetree/bindings/gpu/brcm,bcm-v3d.yaml
index e6485f7b046f..217c42874f41 100644
--- a/Documentation/devicetree/bindings/gpu/brcm,bcm-v3d.yaml
+++ b/Documentation/devicetree/bindings/gpu/brcm,bcm-v3d.yaml
@@ -16,6 +16,7 @@ properties:
 
   compatible:
     enum:
+      - brcm,2711-v3d
       - brcm,7268-v3d
       - brcm,7278-v3d
 

From patchwork Fri Jun  3 09:26:06 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Robinson <pbrobinson@gmail.com>
X-Patchwork-Id: 12868858
Return-Path: 
 <linux-arm-kernel-bounces+linux-arm-kernel=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 38771C433EF
	for <linux-arm-kernel@archiver.kernel.org>;
 Fri,  3 Jun 2022 09:28:40 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=CsFn8xVLllX/9RnoBTFV+s2Q6gWciRpHgjnlM6301Dk=; b=31XkkvbK8oROTV
	9cHP80Vm3+s9RtL+ZnVf2kEN4saia9adOZ1RpaV5+c45Dgd3BF4uob35mtmuYgdfJIuEP6zK7WuZk
	LL4itj/zhTJCxOhZieSmgNZckb2VtBTvoI5JCy6AmJwLX75Ubcwbqtf29urrZxCMq9Ml/A3o3plHp
	025io5B0PC1dTWuiYZJgHSXeFdDgWAY23CTWsNRIRYb41atjTNSrLsvhurg7TdDxVjUgpdNpHAdCV
	WLHHlChEL/HlMrwOm6v1VFFPfsCPuQjZgUdHHyDTXoBQe3ttqJlky4oY10zkYS5aiTPqTP90IFel4
	xv70unukSqWUcKpBubUg==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nx3aX-006oRk-Kf; Fri, 03 Jun 2022 09:27:26 +0000
Received: from mail-ed1-x531.google.com ([2a00:1450:4864:20::531])
	by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nx3ZV-006nub-BS; Fri, 03 Jun 2022 09:26:23 +0000
Received: by mail-ed1-x531.google.com with SMTP id v25so9382164eda.6;
        Fri, 03 Jun 2022 02:26:18 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20210112;
        h=from:to:cc:subject:date:message-id:in-reply-to:references
         :mime-version:content-transfer-encoding;
        bh=s9sCYAt/a4BS6Ai8110s+d7vp0IET2Yh4ADU9eLNXM4=;
        b=I7uK6xIaQGAzmie23EWFc9R+LMmkTOmXOVVZNjOsccpEbpt3FFBkUgS70I4Fygvjex
         bu1xTU9iSsGpMWZojykwGXvX9mnHnYPlggzBDDAd1X3nHDTY2TbEq7mvk21elChncYUW
         tjWQJutHvwFGHo/FHWZntIwIkw+HsVlKgPfZBiiaejLk81Q9dkEXS7wU8SUgx5iwJ31q
         8UN+M1izhvHkMmDYXEZN3Ga5L4NHQ6twcBhE9b9CxT77gmZ+JeYso2ZwAZh3+nWJJcTL
         ADkud7GtsrSyjVuKxEcyeR42d1mLdryKFpWHyag+4DvUjQ+WZk6xGZ4OGaGv7pRLJv/1
         g9/w==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20210112;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=s9sCYAt/a4BS6Ai8110s+d7vp0IET2Yh4ADU9eLNXM4=;
        b=l4zJIO+h1fQTwj11X2yahtqeid9tXg4g8bl8wP4sOve+pPxnEqG2R3nv4VYYsiH6uB
         vR6As4oO8CDJv19LWzWfAGFD10xPdaikMsbSK+LQzv9/qgXPAR6Sn4Kc/wVOw+Mnizrj
         4o1gTiDnf8P7YISi4n4I/95eBoZ/l5NZpVVt24f0g7ZRdxhAeRRaGMvgeN8H6RccYysb
         GFQqQO2dEOms8Krc0pJw932WI+PuN55mwADfvQuFEPpDWTpM435gxz5Qb7RevSRXO4tx
         95zbeTDrdyGsqWIn4IliYniYwFy8iO7ZDOZKh2igKypyYyWlGiBC+FD4lL6k6k0SQw+x
         rFLA==
X-Gm-Message-State: AOAM531MbX1a9h/8HnIhoeY3C3gMYN+iy3Ptig9M6FK7Bhanxafi48h4
	up7PFWjCSQ91uOcgzb1VajA=
X-Google-Smtp-Source: 
 ABdhPJz6hpcDnY/xAYhrLZeKlTfkhMN3sfir2otDWJSfHt9llVhA0AOpRlvfUPA5JvZfOFP/WZKVNg==
X-Received: by 2002:a05:6402:430d:b0:42e:199a:4eb8 with SMTP id
 m13-20020a056402430d00b0042e199a4eb8mr6402367edc.411.1654248376814;
        Fri, 03 Jun 2022 02:26:16 -0700 (PDT)
Received: from morpheus.home.roving-it.com
 (3.e.2.0.0.0.0.0.0.0.0.0.0.0.0.0.1.8.6.2.1.1.b.f.0.b.8.0.1.0.0.2.ip6.arpa.
 [2001:8b0:fb11:2681::2e3])
        by smtp.googlemail.com with ESMTPSA id
 b5-20020a17090691c500b006feaa22e367sm2617672ejx.165.2022.06.03.02.26.15
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Fri, 03 Jun 2022 02:26:16 -0700 (PDT)
From: Peter Robinson <pbrobinson@gmail.com>
To: Arnd Bergmann <arnd@arndb.de>,
	bcm-kernel-feedback-list@broadcom.com,
	Daniel Vetter <daniel@ffwll.ch>,
	David Airlie <airlied@linux.ie>,
	devicetree@vger.kernel.org,
	dri-devel@lists.freedesktop.org,
	Emma Anholt <emma@anholt.net>,
	Florian Fainelli <f.fainelli@gmail.com>,
	javierm@redhat.com,
	linux-arm-kernel@lists.infradead.org,
	linux-rpi-kernel@lists.infradead.org,
	maxime@cerno.tech,
	Nicolas Saenz Julienne <nsaenz@kernel.org>,
	Rob Herring <robh+dt@kernel.org>,
	Stefan Wahren <stefan.wahren@i2se.com>,
	Catalin Marinas <catalin.marinas@arm.com>,
	Will Deacon <will@kernel.org>
Cc: Peter Robinson <pbrobinson@gmail.com>,
	Nicolas Saenz Julienne <nsaenzjulienne@suse.de>
Subject: [PATCH v6 2/6] drm/v3d: Get rid of pm code
Date: Fri,  3 Jun 2022 10:26:06 +0100
Message-Id: <20220603092610.1909675-3-pbrobinson@gmail.com>
X-Mailer: git-send-email 2.36.1
In-Reply-To: <20220603092610.1909675-1-pbrobinson@gmail.com>
References: <20220603092610.1909675-1-pbrobinson@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20220603_022621_444375_6D2B28B8 
X-CRM114-Status: GOOD (  19.38  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+linux-arm-kernel=archiver.kernel.org@lists.infradead.org

Runtime PM doesn't seem to work correctly on this driver. On top of
that, commit 8b6864e3e138 ("drm/v3d/v3d_drv: Remove unused static
variable 'v3d_v3d_pm_ops'") hints that it most likely never did as the
driver's PM ops were not hooked-up.

So, in order to support regular operation with V3D on BCM2711 (Raspberry
Pi 4), get rid of the PM code. PM will be reinstated once we figure out
the underlying issues.

Signed-off-by: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>
Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
Reviewed-by: Javier Martinez Canillas <javierm@redhat.com>
Acked-by: Melissa Wen <mwen@igalia.com>
---
Changes since v4:
- Rebase

Changes since v3:
- Minor updates for rebase

 drivers/gpu/drm/v3d/v3d_debugfs.c | 18 +-----------------
 drivers/gpu/drm/v3d/v3d_drv.c     | 11 -----------
 drivers/gpu/drm/v3d/v3d_gem.c     | 12 +-----------
 3 files changed, 2 insertions(+), 39 deletions(-)

diff --git a/drivers/gpu/drm/v3d/v3d_debugfs.c b/drivers/gpu/drm/v3d/v3d_debugfs.c
index 29fd13109e43..efbde124c296 100644
--- a/drivers/gpu/drm/v3d/v3d_debugfs.c
+++ b/drivers/gpu/drm/v3d/v3d_debugfs.c
@@ -4,7 +4,6 @@
 #include <linux/circ_buf.h>
 #include <linux/ctype.h>
 #include <linux/debugfs.h>
-#include <linux/pm_runtime.h>
 #include <linux/seq_file.h>
 #include <linux/string_helpers.h>
 
@@ -131,11 +130,7 @@ static int v3d_v3d_debugfs_ident(struct seq_file *m, void *unused)
 	struct drm_device *dev = node->minor->dev;
 	struct v3d_dev *v3d = to_v3d_dev(dev);
 	u32 ident0, ident1, ident2, ident3, cores;
-	int ret, core;
-
-	ret = pm_runtime_get_sync(v3d->drm.dev);
-	if (ret < 0)
-		return ret;
+	int core;
 
 	ident0 = V3D_READ(V3D_HUB_IDENT0);
 	ident1 = V3D_READ(V3D_HUB_IDENT1);
@@ -188,9 +183,6 @@ static int v3d_v3d_debugfs_ident(struct seq_file *m, void *unused)
 			   (misccfg & V3D_MISCCFG_OVRTMUOUT) != 0);
 	}
 
-	pm_runtime_mark_last_busy(v3d->drm.dev);
-	pm_runtime_put_autosuspend(v3d->drm.dev);
-
 	return 0;
 }
 
@@ -218,11 +210,6 @@ static int v3d_measure_clock(struct seq_file *m, void *unused)
 	uint32_t cycles;
 	int core = 0;
 	int measure_ms = 1000;
-	int ret;
-
-	ret = pm_runtime_get_sync(v3d->drm.dev);
-	if (ret < 0)
-		return ret;
 
 	if (v3d->ver >= 40) {
 		V3D_CORE_WRITE(core, V3D_V4_PCTR_0_SRC_0_3,
@@ -246,9 +233,6 @@ static int v3d_measure_clock(struct seq_file *m, void *unused)
 		   cycles / (measure_ms * 1000),
 		   (cycles / (measure_ms * 100)) % 10);
 
-	pm_runtime_mark_last_busy(v3d->drm.dev);
-	pm_runtime_put_autosuspend(v3d->drm.dev);
-
 	return 0;
 }
 
diff --git a/drivers/gpu/drm/v3d/v3d_drv.c b/drivers/gpu/drm/v3d/v3d_drv.c
index 1afcd54fbbd5..56d5f831e48b 100644
--- a/drivers/gpu/drm/v3d/v3d_drv.c
+++ b/drivers/gpu/drm/v3d/v3d_drv.c
@@ -19,7 +19,6 @@
 #include <linux/module.h>
 #include <linux/of_platform.h>
 #include <linux/platform_device.h>
-#include <linux/pm_runtime.h>
 #include <linux/reset.h>
 
 #include <drm/drm_drv.h>
@@ -43,7 +42,6 @@ static int v3d_get_param_ioctl(struct drm_device *dev, void *data,
 {
 	struct v3d_dev *v3d = to_v3d_dev(dev);
 	struct drm_v3d_get_param *args = data;
-	int ret;
 	static const u32 reg_map[] = {
 		[DRM_V3D_PARAM_V3D_UIFCFG] = V3D_HUB_UIFCFG,
 		[DRM_V3D_PARAM_V3D_HUB_IDENT1] = V3D_HUB_IDENT1,
@@ -69,17 +67,12 @@ static int v3d_get_param_ioctl(struct drm_device *dev, void *data,
 		if (args->value != 0)
 			return -EINVAL;
 
-		ret = pm_runtime_get_sync(v3d->drm.dev);
-		if (ret < 0)
-			return ret;
 		if (args->param >= DRM_V3D_PARAM_V3D_CORE0_IDENT0 &&
 		    args->param <= DRM_V3D_PARAM_V3D_CORE0_IDENT2) {
 			args->value = V3D_CORE_READ(0, offset);
 		} else {
 			args->value = V3D_READ(offset);
 		}
-		pm_runtime_mark_last_busy(v3d->drm.dev);
-		pm_runtime_put_autosuspend(v3d->drm.dev);
 		return 0;
 	}
 
@@ -280,10 +273,6 @@ static int v3d_platform_drm_probe(struct platform_device *pdev)
 		return -ENOMEM;
 	}
 
-	pm_runtime_use_autosuspend(dev);
-	pm_runtime_set_autosuspend_delay(dev, 50);
-	pm_runtime_enable(dev);
-
 	ret = v3d_gem_init(drm);
 	if (ret)
 		goto dma_free;
diff --git a/drivers/gpu/drm/v3d/v3d_gem.c b/drivers/gpu/drm/v3d/v3d_gem.c
index 92bc0faee84f..7026214a09f0 100644
--- a/drivers/gpu/drm/v3d/v3d_gem.c
+++ b/drivers/gpu/drm/v3d/v3d_gem.c
@@ -6,7 +6,6 @@
 #include <linux/io.h>
 #include <linux/module.h>
 #include <linux/platform_device.h>
-#include <linux/pm_runtime.h>
 #include <linux/reset.h>
 #include <linux/sched/signal.h>
 #include <linux/uaccess.h>
@@ -367,9 +366,6 @@ v3d_job_free(struct kref *ref)
 	dma_fence_put(job->irq_fence);
 	dma_fence_put(job->done_fence);
 
-	pm_runtime_mark_last_busy(job->v3d->drm.dev);
-	pm_runtime_put_autosuspend(job->v3d->drm.dev);
-
 	if (job->perfmon)
 		v3d_perfmon_put(job->perfmon);
 
@@ -471,14 +467,10 @@ v3d_job_init(struct v3d_dev *v3d, struct drm_file *file_priv,
 	job->v3d = v3d;
 	job->free = free;
 
-	ret = pm_runtime_get_sync(v3d->drm.dev);
-	if (ret < 0)
-		goto fail;
-
 	ret = drm_sched_job_init(&job->base, &v3d_priv->sched_entity[queue],
 				 v3d_priv);
 	if (ret)
-		goto fail_job;
+		goto fail;
 
 	if (has_multisync) {
 		if (se->in_sync_count && se->wait_stage == queue) {
@@ -509,8 +501,6 @@ v3d_job_init(struct v3d_dev *v3d, struct drm_file *file_priv,
 
 fail_deps:
 	drm_sched_job_cleanup(&job->base);
-fail_job:
-	pm_runtime_put_autosuspend(v3d->drm.dev);
 fail:
 	kfree(*container);
 	*container = NULL;

From patchwork Fri Jun  3 09:26:07 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Robinson <pbrobinson@gmail.com>
X-Patchwork-Id: 12868857
Return-Path: 
 <linux-arm-kernel-bounces+linux-arm-kernel=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id E3482C43334
	for <linux-arm-kernel@archiver.kernel.org>;
 Fri,  3 Jun 2022 09:28:34 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=8KzIyMrrZn2ethrc5NrtuSyV5CfdoShCfiRqNzcBPwQ=; b=YbiZ9M5D99zNht
	S1H28sanXD7hwS4aOmS3Vh9vaLB7/T8t+MEoYfbp/8iYqPr6EX7sXQuM3BQZu69/tZyXfhTbiVoIJ
	8V4Utk0n1qkoZpQdCQfD0LaKqrNPs+UuVUiS+NSZXPsyw3fUSNLCmUc96STp7hBSqj7NUakXCXF8W
	wYDCHxoVyiQt+2Hor7AjYViSOWDUyoCdrZExj25JypJmZlrvlIIX0sFBHFBsxAzgGe33AhLzWvNTw
	N/afSYoMwLIrHqjEOUIPoQ2b55+FRWspuNz2XdkvhYrUltLjuqjFlFR41FT1cqcpmLoiH32+G4LPw
	ZkSYoBPyQMbLcGO9jdSA==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nx3aB-006oIv-80; Fri, 03 Jun 2022 09:27:03 +0000
Received: from mail-ed1-x52c.google.com ([2a00:1450:4864:20::52c])
	by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nx3ZV-006nvD-BI; Fri, 03 Jun 2022 09:26:23 +0000
Received: by mail-ed1-x52c.google.com with SMTP id c2so9373834edf.5;
        Fri, 03 Jun 2022 02:26:18 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20210112;
        h=from:to:cc:subject:date:message-id:in-reply-to:references
         :mime-version:content-transfer-encoding;
        bh=3QIoQb5QbRVNOs1RbxwOuYWdQ4RolJT8xZ/PEN472xw=;
        b=lmCyckO1JfhORtymrMAw3PdU4nq7PrawiqPgxXxYW7btxcfTfx8+r2WRvAqKrEervY
         tEPe2LGnx5LLR2IAGnYFAxsP2e7gMQlNFEVqh4GraTlnLdkgYM5QohEs7R6xDxGu/D/H
         RRXGyzy+Y4QoVQXpz6a13yjRzkjOMz4UYzT8Zd5ewzOkDQB8ONr1rovYtJ4m9IDCqAwG
         GLwZnEPuKTsxjGcDz/th8HLINjbe7nq4eoOcqMOqReZMHVVATirjLgocmKh2LZrbSP1E
         MvKw0e9ciET/2v6CeS9Qz//5UkV9e4iPu+eQSkgLaltIP+qh+1QCKcQBf6oBQ/kWDaAu
         R+eA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20210112;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=3QIoQb5QbRVNOs1RbxwOuYWdQ4RolJT8xZ/PEN472xw=;
        b=nFSAx+AzNx6pFi9dz5ir3HClWVgcYprrtR6BDBgYRTlpvbh8jmyTyKXC8uzGbEJhm/
         sbHXYrOXacQqkvdB7VfYbGTujZQr2Y/e9F15SqLNyUv8k7t71an6hVXz1Z/MfnXcfS82
         Vps8TjwZ6p+RcrWPzMVxDWXtvMxAPRGnZ5gdS3jXxQQqYyGmUiyNQpzPBtuh3UF787Uj
         qOAxzjUc4yBZbQq6h0ercjDjCSQyanGCTW1npJW4Qdt8WI9hdx/FKHF4hZ2v/iMI1ywK
         0paGyyntTIs7d/XHKn8eU6gToKrBtcAumfFtXgSElJ+x5dXZjgo7FfQG6HBwZKGziCV1
         2SIA==
X-Gm-Message-State: AOAM531VYerJpxGIV702sWU+Yo68GinGShtVeTMbvuS9Te/R648nVEVM
	bPNAXXHcipyBjMHHCOMH0kE=
X-Google-Smtp-Source: 
 ABdhPJwWDSaSdYoQBx6iHBJAm8vaTsVDbYmRXyOGiD0pUsPbS9VGBLoLVtY7Ji3bYn69DPVycbuvKA==
X-Received: by 2002:a05:6402:228d:b0:42d:e319:7297 with SMTP id
 cw13-20020a056402228d00b0042de3197297mr9641542edb.79.1654248377816;
        Fri, 03 Jun 2022 02:26:17 -0700 (PDT)
Received: from morpheus.home.roving-it.com
 (3.e.2.0.0.0.0.0.0.0.0.0.0.0.0.0.1.8.6.2.1.1.b.f.0.b.8.0.1.0.0.2.ip6.arpa.
 [2001:8b0:fb11:2681::2e3])
        by smtp.googlemail.com with ESMTPSA id
 b5-20020a17090691c500b006feaa22e367sm2617672ejx.165.2022.06.03.02.26.16
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Fri, 03 Jun 2022 02:26:17 -0700 (PDT)
From: Peter Robinson <pbrobinson@gmail.com>
To: Arnd Bergmann <arnd@arndb.de>,
	bcm-kernel-feedback-list@broadcom.com,
	Daniel Vetter <daniel@ffwll.ch>,
	David Airlie <airlied@linux.ie>,
	devicetree@vger.kernel.org,
	dri-devel@lists.freedesktop.org,
	Emma Anholt <emma@anholt.net>,
	Florian Fainelli <f.fainelli@gmail.com>,
	javierm@redhat.com,
	linux-arm-kernel@lists.infradead.org,
	linux-rpi-kernel@lists.infradead.org,
	maxime@cerno.tech,
	Nicolas Saenz Julienne <nsaenz@kernel.org>,
	Rob Herring <robh+dt@kernel.org>,
	Stefan Wahren <stefan.wahren@i2se.com>,
	Catalin Marinas <catalin.marinas@arm.com>,
	Will Deacon <will@kernel.org>
Cc: Peter Robinson <pbrobinson@gmail.com>,
	Nicolas Saenz Julienne <nsaenzjulienne@suse.de>
Subject: [PATCH v6 3/6] drm/v3d: Add support for bcm2711
Date: Fri,  3 Jun 2022 10:26:07 +0100
Message-Id: <20220603092610.1909675-4-pbrobinson@gmail.com>
X-Mailer: git-send-email 2.36.1
In-Reply-To: <20220603092610.1909675-1-pbrobinson@gmail.com>
References: <20220603092610.1909675-1-pbrobinson@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20220603_022621_436848_AE475253 
X-CRM114-Status: GOOD (  15.48  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+linux-arm-kernel=archiver.kernel.org@lists.infradead.org

Add compatible string and Kconfig options and help for bcm2711.

Signed-off-by: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>
Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
Reviewed-by: Stefan Wahren <stefan.wahren@i2se.com>
Reviewed-by: Javier Martinez Canillas <javierm@redhat.com>
---
Changes since v5:
- Update help text to cover all supported SoCs

Changes since v4:
- Change compatible to align downstream and other HW, reorder to suit

 drivers/gpu/drm/v3d/Kconfig   | 5 +++--
 drivers/gpu/drm/v3d/v3d_drv.c | 1 +
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/v3d/Kconfig b/drivers/gpu/drm/v3d/Kconfig
index e973ec487484..ce62c5908e1d 100644
--- a/drivers/gpu/drm/v3d/Kconfig
+++ b/drivers/gpu/drm/v3d/Kconfig
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0-only
 config DRM_V3D
 	tristate "Broadcom V3D 3.x and newer"
-	depends on ARCH_BCM || ARCH_BRCMSTB || COMPILE_TEST
+	depends on ARCH_BCM || ARCH_BRCMSTB || ARCH_BCM2835 || COMPILE_TEST
 	depends on DRM
 	depends on COMMON_CLK
 	depends on MMU
@@ -9,4 +9,5 @@ config DRM_V3D
 	select DRM_GEM_SHMEM_HELPER
 	help
 	  Choose this option if you have a system that has a Broadcom
-	  V3D 3.x or newer GPU, such as BCM7268.
+	  V3D 3.x or newer GPUs. SoCs supported include the BCM2711,
+	  BCM7268 and BCM7278.
diff --git a/drivers/gpu/drm/v3d/v3d_drv.c b/drivers/gpu/drm/v3d/v3d_drv.c
index 56d5f831e48b..8c7f910daa28 100644
--- a/drivers/gpu/drm/v3d/v3d_drv.c
+++ b/drivers/gpu/drm/v3d/v3d_drv.c
@@ -191,6 +191,7 @@ static const struct drm_driver v3d_drm_driver = {
 };
 
 static const struct of_device_id v3d_of_match[] = {
+	{ .compatible = "brcm,2711-v3d" },
 	{ .compatible = "brcm,7268-v3d" },
 	{ .compatible = "brcm,7278-v3d" },
 	{},

From patchwork Fri Jun  3 09:26:08 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Robinson <pbrobinson@gmail.com>
X-Patchwork-Id: 12868859
Return-Path: 
 <linux-arm-kernel-bounces+linux-arm-kernel=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 18B9EC433EF
	for <linux-arm-kernel@archiver.kernel.org>;
 Fri,  3 Jun 2022 09:29:30 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=Wex8xlPuDcoN9SM3Q2JNmJfHJWSyEG3yxiTwSWZr1hc=; b=JvZdK8fhEWJndO
	eIy4JVH8pKth0NJx09vO4YscHku/GBMC4EmpbMI8h27NlihUwS8iOquqsagaVWIy1xBh7eCKQQaUI
	tIPCz9iJdvMk8qk0QgUMvPWbRScMET2bvky2JSkw5ESTXKaY1cdJWJKq8/7L/24fjWWc6FcQwrMCc
	109GMSms8NZxlnLENtAiFZBhz/ysaHJM2sDKaDUHDfCQvlpmqe9xEIXQHLZrpuofiV6csrEV7ZCtq
	IBrm8RXkfAoTRh2RDFJ+IYxMTzObMC8lRUrNQ9C0735miNepecOgu0DsCGo2mdw554ydeqcnr6pBh
	swob9cnnT2Q7pNG7tLig==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nx3aw-006od0-MM; Fri, 03 Jun 2022 09:27:52 +0000
Received: from mail-ej1-x62a.google.com ([2a00:1450:4864:20::62a])
	by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nx3ZY-006nwJ-87; Fri, 03 Jun 2022 09:26:25 +0000
Received: by mail-ej1-x62a.google.com with SMTP id v1so4097216ejg.13;
        Fri, 03 Jun 2022 02:26:20 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20210112;
        h=from:to:cc:subject:date:message-id:in-reply-to:references
         :mime-version:content-transfer-encoding;
        bh=ul5kl3xkOBddCK6BZmBUL37TSKlFrzs4FlZ0lPWju6E=;
        b=Ois8e1psuQKU/vL+ZZ2WfE2Yt78mnui0q5PLO8u67c9bYrB1rcazPSj9flPXInIXii
         nBZlzOcBLDpYGo4rgvLTRuaFGgWIj47malQftngol/m/MWpBoorH9x5vd91bv+KO8FJJ
         qdHvvLj3a6CVqzlNOaI1HKmIDYpUCgr0JT3hLOet0Ewvkv5UDbCjhN0cZiMZQYD6aD8p
         37WQc2Mm8aPK8EnhFhlHKqC+1CbEj3SjdcadoU1Qx59b1qH0NhS2R6c89WCMsAe0duOi
         l04FubgcTsL/M108tDRa2mSkuQ24+v9Qfd5LZaeLvgjLaGiqN3e9xIbKrDFW06Orpxjk
         ohmQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20210112;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=ul5kl3xkOBddCK6BZmBUL37TSKlFrzs4FlZ0lPWju6E=;
        b=cw3TgQ3U/nX8lf/2s1njdS56LXygXZRUuTRZUs9xzNKAz07/ZAlclMG1iZFG4oLUFA
         A3QSzyx691SmVHDw/QWYUSYnyJ3r9zYMmbwtDID2J/vQBB/shCtsVC+VjAyCn4/RxDAu
         ywLxbM0cvmkgBFW+VG3uOU26f/BlezqYbrmDrXaZA8nzz96QFM9VgTJTZwkCgZHm/ehc
         u6ek3HjDNe5MYMjWC1nKt+3jJCKbn8Hpr9NtFoK2nrimRjwtabNzLXRdj8KJ6lleTz37
         TXJ0y2CjjdVqwhm01RpIgxe4kX2mB//D9gGARAjr5RSqSsDmAeXM7IzVVMs+79/ySsJP
         7PNg==
X-Gm-Message-State: AOAM531xHgfYBxuSuO4UY2NY9xw3Dqb4xSZU5unGDrTh+6AkxP/U9sJy
	/JA2yy+dR4LIOW/QQsh5eqM=
X-Google-Smtp-Source: 
 ABdhPJyB/WV+eEvN1uLE1B454uXjO9+6jxoRuQ9KICIHvdfB1Hz+zTnHqm6rnXfD6SFeB+ZE8ahgUw==
X-Received: by 2002:a17:907:a406:b0:703:db59:35a with SMTP id
 sg6-20020a170907a40600b00703db59035amr7723403ejc.321.1654248379038;
        Fri, 03 Jun 2022 02:26:19 -0700 (PDT)
Received: from morpheus.home.roving-it.com
 (3.e.2.0.0.0.0.0.0.0.0.0.0.0.0.0.1.8.6.2.1.1.b.f.0.b.8.0.1.0.0.2.ip6.arpa.
 [2001:8b0:fb11:2681::2e3])
        by smtp.googlemail.com with ESMTPSA id
 b5-20020a17090691c500b006feaa22e367sm2617672ejx.165.2022.06.03.02.26.18
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Fri, 03 Jun 2022 02:26:18 -0700 (PDT)
From: Peter Robinson <pbrobinson@gmail.com>
To: Arnd Bergmann <arnd@arndb.de>,
	bcm-kernel-feedback-list@broadcom.com,
	Daniel Vetter <daniel@ffwll.ch>,
	David Airlie <airlied@linux.ie>,
	devicetree@vger.kernel.org,
	dri-devel@lists.freedesktop.org,
	Emma Anholt <emma@anholt.net>,
	Florian Fainelli <f.fainelli@gmail.com>,
	javierm@redhat.com,
	linux-arm-kernel@lists.infradead.org,
	linux-rpi-kernel@lists.infradead.org,
	maxime@cerno.tech,
	Nicolas Saenz Julienne <nsaenz@kernel.org>,
	Rob Herring <robh+dt@kernel.org>,
	Stefan Wahren <stefan.wahren@i2se.com>,
	Catalin Marinas <catalin.marinas@arm.com>,
	Will Deacon <will@kernel.org>
Cc: Peter Robinson <pbrobinson@gmail.com>,
	Nicolas Saenz Julienne <nsaenzjulienne@suse.de>
Subject: [PATCH v6 4/6] ARM: dts: bcm2711: Enable V3D
Date: Fri,  3 Jun 2022 10:26:08 +0100
Message-Id: <20220603092610.1909675-5-pbrobinson@gmail.com>
X-Mailer: git-send-email 2.36.1
In-Reply-To: <20220603092610.1909675-1-pbrobinson@gmail.com>
References: <20220603092610.1909675-1-pbrobinson@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20220603_022624_462333_884DCEEE 
X-CRM114-Status: GOOD (  11.62  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+linux-arm-kernel=archiver.kernel.org@lists.infradead.org

This adds the entry for V3D for bcm2711 (used in the Raspberry Pi 4)
and the associated firmware clock entry.

Signed-off-by: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>
Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
Reviewed-by: Javier Martinez Canillas <javierm@redhat.com>
---
Changes since v5:
- Update the compatible to match the other updated ones

Changes since v4:
- Move the firmware clock to bcm2711-rpi.dtsi

 arch/arm/boot/dts/bcm2711-rpi.dtsi |  4 ++++
 arch/arm/boot/dts/bcm2711.dtsi     | 11 +++++++++++
 2 files changed, 15 insertions(+)

diff --git a/arch/arm/boot/dts/bcm2711-rpi.dtsi b/arch/arm/boot/dts/bcm2711-rpi.dtsi
index ca266c5d9f9b..98817a6675b9 100644
--- a/arch/arm/boot/dts/bcm2711-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2711-rpi.dtsi
@@ -69,6 +69,10 @@ blconfig: nvram@0 {
 	};
 };
 
+&v3d {
+	clocks = <&firmware_clocks 5>;
+};
+
 &vchiq {
 	interrupts = <GIC_SPI 34 IRQ_TYPE_LEVEL_HIGH>;
 };
diff --git a/arch/arm/boot/dts/bcm2711.dtsi b/arch/arm/boot/dts/bcm2711.dtsi
index 89af57482bc8..20e6771e8b1f 100644
--- a/arch/arm/boot/dts/bcm2711.dtsi
+++ b/arch/arm/boot/dts/bcm2711.dtsi
@@ -601,6 +601,17 @@ genet_mdio: mdio@e14 {
 				#size-cells = <0x0>;
 			};
 		};
+
+		v3d: gpu@7ec00000 {
+			compatible = "brcm,2711-v3d";
+			reg = <0x0 0x7ec00000 0x4000>,
+			      <0x0 0x7ec04000 0x4000>;
+			reg-names = "hub", "core0";
+
+			power-domains = <&pm BCM2835_POWER_DOMAIN_GRAFX_V3D>;
+			resets = <&pm BCM2835_RESET_V3D>;
+			interrupts = <GIC_SPI 74 IRQ_TYPE_LEVEL_HIGH>;
+		};
 	};
 };
 

From patchwork Fri Jun  3 09:26:09 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Robinson <pbrobinson@gmail.com>
X-Patchwork-Id: 12868860
Return-Path: 
 <linux-arm-kernel-bounces+linux-arm-kernel=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 241D2C433EF
	for <linux-arm-kernel@archiver.kernel.org>;
 Fri,  3 Jun 2022 09:29:46 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=co38PxvFFiBOsdSI10pr2N8kfRRMe4tcyMS08R2HT6g=; b=aRNRvFz8h+nwo1
	BFr4pp4lc3GqbnwrD6r4LOfQXT+T4pyA+aW8oXCaPyjIo0n8mrJqTlcs8YkRojb8PxtNJ++P+hKN4
	w8M8NY34i/D1d7829Uu9SnLW4eXRhyGiWsWSiq/y4LClic8KKVjoA8NsZzkHUs+WmrUmP+xXBUsy0
	3dsQaPv6znXzjVatmhI3qUqc6SWJdrgSaPnGgBHHeNWcTBY83WCnbSLDC+tT3sy2F1CehA5qUDtT6
	tGqWuaSxASY1lCQITkLOJJlqoRp4uAr4Ev9r2ZjFddlIaEBGQ08PbH+1uMUxswGCtInX8OPvpawpT
	W9PH4qMw5ddQlgjPMsMA==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nx3ba-006oty-UB; Fri, 03 Jun 2022 09:28:31 +0000
Received: from mail-ej1-x629.google.com ([2a00:1450:4864:20::629])
	by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nx3ZY-006nxM-Tm; Fri, 03 Jun 2022 09:26:27 +0000
Received: by mail-ej1-x629.google.com with SMTP id h23so3708878ejj.12;
        Fri, 03 Jun 2022 02:26:21 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20210112;
        h=from:to:cc:subject:date:message-id:in-reply-to:references
         :mime-version:content-transfer-encoding;
        bh=ssOJLbGbegD1K46c6tLMgIHQD6IP233xl23vf6Dwrk8=;
        b=ORfIo6ijxncBWoW1zXqygp5d/hED615KbJsb0wAKIXFVm9YI7v2rGtaxqmLpUYFs5+
         0QwcxMDNKTQ0nynbMf1nZ0attlc+BIe6HuKb+nSviTbQDLX/kycf/lna6EhBH3C+IaWT
         fP4yQtrqjsB6H0KXcPMBeg7l8mN5oGq0NytiCc3pOv5PKUV5JOQKbtRABbm3kTLTkHJy
         K6krOmT26OsrDjQaQazX9rMWMgqYtWwQQLn4C7d85pH7GgwIBUASsXRRHKwvgb9sFdJl
         pDjJHA5QIJiR6xePJTgk57UE0sy46OHgIb3p2MD1WbeepIGW2ZezisrrXYSMS7l1KWP0
         CSBQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20210112;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=ssOJLbGbegD1K46c6tLMgIHQD6IP233xl23vf6Dwrk8=;
        b=XlIl3ppeSou6CoswSc+KBhhPyrg4LzzQDpWkZuQUzlXF1qUgn3pE+3fm+cCKHxPmGr
         Z8BSgFLgTbZSAdiFj1RvQS5ymq/6nGTiiwoRh0ZVHn4o7yBoxf0gGlKVjrvvFTZJrQau
         o9uyr3nVjwyAA0xOnHYHzmEVI4oayDSqY6frZR93eOd4s0flSQ6Mrnnj26HeONjXEtJa
         fIBmJULoBA/F2DQqdtnfeTMy7kqcIEMXq3wb9mN+lRKy7n12xF1RCFdK1pDPUDqtcSL9
         at0QwRC2osrwf//qjs3bgxkNxEB1/1VTxrhrW6Gh74wsw8TskrQgq4zatBTJvPlcA1By
         HZfg==
X-Gm-Message-State: AOAM532C9KqtesCU+UK2Mi0JHajko4ogrD5XT5PG+dnX7itKbb3obK5x
	RqoXTzo+j7vo+Os2l5zPoL4=
X-Google-Smtp-Source: 
 ABdhPJxT3uTOHv8dOKf6vo3zopR23+u+e+tO/4ZN3QFTs3V+EAqg3wHOSloOBT6vRSAh79itOUzGew==
X-Received: by 2002:a17:906:4785:b0:6fe:f8ac:2494 with SMTP id
 cw5-20020a170906478500b006fef8ac2494mr7667931ejc.199.1654248380180;
        Fri, 03 Jun 2022 02:26:20 -0700 (PDT)
Received: from morpheus.home.roving-it.com
 (3.e.2.0.0.0.0.0.0.0.0.0.0.0.0.0.1.8.6.2.1.1.b.f.0.b.8.0.1.0.0.2.ip6.arpa.
 [2001:8b0:fb11:2681::2e3])
        by smtp.googlemail.com with ESMTPSA id
 b5-20020a17090691c500b006feaa22e367sm2617672ejx.165.2022.06.03.02.26.19
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Fri, 03 Jun 2022 02:26:19 -0700 (PDT)
From: Peter Robinson <pbrobinson@gmail.com>
To: Arnd Bergmann <arnd@arndb.de>,
	bcm-kernel-feedback-list@broadcom.com,
	Daniel Vetter <daniel@ffwll.ch>,
	David Airlie <airlied@linux.ie>,
	devicetree@vger.kernel.org,
	dri-devel@lists.freedesktop.org,
	Emma Anholt <emma@anholt.net>,
	Florian Fainelli <f.fainelli@gmail.com>,
	javierm@redhat.com,
	linux-arm-kernel@lists.infradead.org,
	linux-rpi-kernel@lists.infradead.org,
	maxime@cerno.tech,
	Nicolas Saenz Julienne <nsaenz@kernel.org>,
	Rob Herring <robh+dt@kernel.org>,
	Stefan Wahren <stefan.wahren@i2se.com>,
	Catalin Marinas <catalin.marinas@arm.com>,
	Will Deacon <will@kernel.org>
Cc: Peter Robinson <pbrobinson@gmail.com>,
	Nicolas Saenz Julienne <nsaenzjulienne@suse.de>
Subject: [PATCH v6 5/6] ARM: configs: Enable DRM_V3D
Date: Fri,  3 Jun 2022 10:26:09 +0100
Message-Id: <20220603092610.1909675-6-pbrobinson@gmail.com>
X-Mailer: git-send-email 2.36.1
In-Reply-To: <20220603092610.1909675-1-pbrobinson@gmail.com>
References: <20220603092610.1909675-1-pbrobinson@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20220603_022625_043881_67CDA462 
X-CRM114-Status: GOOD (  11.11  )
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+linux-arm-kernel=archiver.kernel.org@lists.infradead.org

BCM2711, the SoC used on the Raspberry Pi 4 has a different 3D
render GPU IP than its predecessors. Enable it it on multi v7
and bcm2835 configs.

Signed-off-by: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>
Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
Reviewed-by: Stefan Wahren <stefan.wahren@i2se.com>
Reviewed-by: Javier Martinez Canillas <javierm@redhat.com>
---
Changes since v4:
- Added to bcm2835_defconfig

 arch/arm/configs/bcm2835_defconfig  | 1 +
 arch/arm/configs/multi_v7_defconfig | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/arm/configs/bcm2835_defconfig b/arch/arm/configs/bcm2835_defconfig
index a9ed79b7f871..9270512c14ea 100644
--- a/arch/arm/configs/bcm2835_defconfig
+++ b/arch/arm/configs/bcm2835_defconfig
@@ -106,6 +106,7 @@ CONFIG_REGULATOR_GPIO=y
 CONFIG_MEDIA_SUPPORT=y
 CONFIG_MEDIA_CAMERA_SUPPORT=y
 CONFIG_DRM=y
+CONFIG_DRM_V3D=y
 CONFIG_DRM_VC4=y
 CONFIG_FB_SIMPLE=y
 CONFIG_FRAMEBUFFER_CONSOLE=y
diff --git a/arch/arm/configs/multi_v7_defconfig b/arch/arm/configs/multi_v7_defconfig
index d6a6811f0539..e2db5cdc66b7 100644
--- a/arch/arm/configs/multi_v7_defconfig
+++ b/arch/arm/configs/multi_v7_defconfig
@@ -731,6 +731,7 @@ CONFIG_DRM_IMX_PARALLEL_DISPLAY=m
 CONFIG_DRM_IMX_TVE=m
 CONFIG_DRM_IMX_LDB=m
 CONFIG_DRM_IMX_HDMI=m
+CONFIG_DRM_V3D=m
 CONFIG_DRM_VC4=m
 CONFIG_DRM_ETNAVIV=m
 CONFIG_DRM_MXSFB=m

From patchwork Fri Jun  3 09:26:10 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peter Robinson <pbrobinson@gmail.com>
X-Patchwork-Id: 12868861
Return-Path: 
 <linux-arm-kernel-bounces+linux-arm-kernel=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 6E631C43334
	for <linux-arm-kernel@archiver.kernel.org>;
 Fri,  3 Jun 2022 09:30:51 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=ewalPF+DwARVsN56KhB5N1tHwMJzhvoFCl0NNq53tiY=; b=xSvjsLjHt/TOTZ
	W/CNT3jBOIbk7AxnKIGA1dDRFj8EOjfGiY0QGJiPRNERCYNnGLL1rVkSKLIDBO68XA45kp/DJ9AyN
	yhW0mRxKzag4NOHRFWWyTwK3mwKwvV1lmFsVYp5c5E13MFsUcAX4F8+IoLpZCySZC8QTAS3IpC5dG
	7m/LI1S0ePKc2Fhn8trlkFL41VyGI02mvf7La4AsZcpOZYVOK87GSEKgFuODSAF+WOv7SXelRo+RC
	a8hIjowMD+ffqdE3b9UA/EQA8/VHqZjyXQJepvlwi82X2Ry+Xknz5cJ53dnAm0scp3nxksA0B02+v
	OuDUP4+w+RFEIKr7t4/Q==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nx3c9-006p8G-Ea; Fri, 03 Jun 2022 09:29:05 +0000
Received: from mail-ej1-x62c.google.com ([2a00:1450:4864:20::62c])
	by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
	id 1nx3ZZ-006nyZ-EE; Fri, 03 Jun 2022 09:26:27 +0000
Received: by mail-ej1-x62c.google.com with SMTP id q21so14820064ejm.1;
        Fri, 03 Jun 2022 02:26:22 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20210112;
        h=from:to:cc:subject:date:message-id:in-reply-to:references
         :mime-version:content-transfer-encoding;
        bh=1zf2KrerTAi6PDusMTCodCD/1wt7NqOtkOzcZDBwMCw=;
        b=XE9UmS5uUXbk7dW1oDHNECJIhdydnEwr/sWPOzRw9ppFwzYiHS0APSXiubukvYTcYI
         b522fiw0Gw7g6CtcBw8RNd27HJ8E0hm99kzeHumJ/MqnpBZR0GNBXWywp4SXWyB4Gr2O
         qPe4LPG5u8V+1PFfVYcG9fiM5sLOgs9A5s9mqZwloKOue3TwMNBlYENIpiVCtmY/stRB
         gonkzG2vv6NQDP1zv2AA191Fz2UuWQ4RbSGq4pJuV19N42oW/m1/s+JazlgrQxIudnuA
         uyCiJ+88fvPAa4vW7s1t+RW/SfRTNH6xlFSjxtbeCFn+EeYMDEOSwYXwjb3L5wCs2kMm
         rmAg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20210112;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=1zf2KrerTAi6PDusMTCodCD/1wt7NqOtkOzcZDBwMCw=;
        b=aZafyQpUWyo3HqIvEXanWly5Efg5SoAkeqejQF/DxZsBLkPNYs05JqMXonCwP7hvzi
         Oq7r4LEt7HXO0br7N2ptLsRAW7ZjVFtt4XpeTXi16NjUDOuufhNfLjM3Gl1MhuueVbDH
         WQdTFbNmuvt84DSMdCp/HTOX57LIsI6aswacZ4hF7cXOkGiH7LdxMGXuipUt2I1o6P8i
         lLav2+n2OEiSBNbVmXRwM3DV6FoelzWfG2b2FJFtLeaSx2cAoRl3VJ6kYFNMNLWXmodr
         RLRxPT30W6LPULx8j9THZtvxXuX0T9Oigsvma9Hia7zQe4Nkkzf/FSFcVGvEdsG939g9
         rgzg==
X-Gm-Message-State: AOAM5313rUDrKgwcMSqH/yuTqv9mZOv1fEK+MGivgTT0dLZLX7McebwB
	gwMquu99bY3IoZRei9dibYI=
X-Google-Smtp-Source: 
 ABdhPJym7Lt4blXmNu6IRUD0ciqXO8Zaa052+4cPkXGXxp1oFtfUf2t6+BktugcotPe8SSwqsoKgOg==
X-Received: by 2002:a17:906:6a28:b0:6ff:d8b:955e with SMTP id
 qw40-20020a1709066a2800b006ff0d8b955emr7970055ejc.505.1654248381223;
        Fri, 03 Jun 2022 02:26:21 -0700 (PDT)
Received: from morpheus.home.roving-it.com
 (3.e.2.0.0.0.0.0.0.0.0.0.0.0.0.0.1.8.6.2.1.1.b.f.0.b.8.0.1.0.0.2.ip6.arpa.
 [2001:8b0:fb11:2681::2e3])
        by smtp.googlemail.com with ESMTPSA id
 b5-20020a17090691c500b006feaa22e367sm2617672ejx.165.2022.06.03.02.26.20
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Fri, 03 Jun 2022 02:26:20 -0700 (PDT)
From: Peter Robinson <pbrobinson@gmail.com>
To: Arnd Bergmann <arnd@arndb.de>,
	bcm-kernel-feedback-list@broadcom.com,
	Daniel Vetter <daniel@ffwll.ch>,
	David Airlie <airlied@linux.ie>,
	devicetree@vger.kernel.org,
	dri-devel@lists.freedesktop.org,
	Emma Anholt <emma@anholt.net>,
	Florian Fainelli <f.fainelli@gmail.com>,
	javierm@redhat.com,
	linux-arm-kernel@lists.infradead.org,
	linux-rpi-kernel@lists.infradead.org,
	maxime@cerno.tech,
	Nicolas Saenz Julienne <nsaenz@kernel.org>,
	Rob Herring <robh+dt@kernel.org>,
	Stefan Wahren <stefan.wahren@i2se.com>,
	Catalin Marinas <catalin.marinas@arm.com>,
	Will Deacon <will@kernel.org>
Cc: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>,
	Peter Robinson <pbrobinson@gmail.com>
Subject: [PATCH v6 6/6] arm64: config: Enable DRM_V3D
Date: Fri,  3 Jun 2022 10:26:10 +0100
Message-Id: <20220603092610.1909675-7-pbrobinson@gmail.com>
X-Mailer: git-send-email 2.36.1
In-Reply-To: <20220603092610.1909675-1-pbrobinson@gmail.com>
References: <20220603092610.1909675-1-pbrobinson@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20220603_022625_530267_CB012BAA 
X-CRM114-Status: UNSURE (   9.93  )
X-CRM114-Notice: Please train this message.
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+linux-arm-kernel=archiver.kernel.org@lists.infradead.org

From: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>

BCM2711, the SoC used on the Raspberry Pi 4 has a different GPU than its
predecessors. Enable it.

Signed-off-by: Nicolas Saenz Julienne <nsaenzjulienne@suse.de>
Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
Reviewed-by: Stefan Wahren <stefan.wahren@i2se.com>
Reviewed-by: Javier Martinez Canillas <javierm@redhat.com>
---
 arch/arm64/configs/defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/configs/defconfig b/arch/arm64/configs/defconfig
index 50aa3d75ab4f..446bac1ef774 100644
--- a/arch/arm64/configs/defconfig
+++ b/arch/arm64/configs/defconfig
@@ -738,6 +738,7 @@ CONFIG_DRM_I2C_ADV7511_AUDIO=y
 CONFIG_DRM_DW_HDMI_AHB_AUDIO=m
 CONFIG_DRM_DW_HDMI_CEC=m
 CONFIG_DRM_IMX_DCSS=m
+CONFIG_DRM_V3D=m
 CONFIG_DRM_VC4=m
 CONFIG_DRM_ETNAVIV=m
 CONFIG_DRM_HISI_HIBMC=m
